### PR TITLE
Finish task edit user & list user

### PR DESCRIPTION
**Pull request by tqtrung09**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, only: [:show]
+  before_action :logged_in_user, only: [:show, :edit, :update, :index]
```
Comment (hellovietnam93) [inline app/controllers/users_controller.rb]: except

File: app/views/users/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  <%= f.label :name %>
+  <%= f.text_field :name, class: "form-control" %>
+
+  <%= f.label :email %>
+  <%= f.email_field :email, class: "form-control" %>
+
+  <%= f.label :password %>
+  <%= f.password_field :password, class: "form-control" %>
+
+  <%= f.label :password_confirmation, "Confirmation" %>
```
Comment (hellovietnam93) [inline app/views/users/_form.html.erb]: I18n

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<% provide :title, t("user.edit") %>
+<% provide(:button_text, "Save changes") %>
```
Comment (hellovietnam93) [inline app/views/users/edit.html.erb]: remove ()

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+<h1><%= t "user.edit" %></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= render "form" %>
+    <div class="gravatar_edit">
+      <%= @user.avatar? ? image_tag(@user.avatar.url) : gravatar_for(@user) %>
```
Comment (hellovietnam93) [inline app/views/users/edit.html.erb]: write helper method to call

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -1,29 +1,9 @@
 <% provide :title, t("user.signup") %>
+<% provide(:button_text, "Create my account") %>
```
Comment (hellovietnam93) [inline app/views/users/new.html.erb]: I18n

**Comment by hellovietnam93** [comment]
OK

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -1,5 +1,10 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, only: [:show]
+  before_action :logged_in_user, except: [:new, :create, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+
+  def index
+    @users = User.all
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: pagination

File: app/views/users/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<% provide(:title, t("user.index")) %>
```
Comment (dieunb) [inline app/views/users/index.html.erb]: remove `()`

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -1,5 +1,10 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, only: [:show]
+  before_action :logged_in_user, except: [:new, :create, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+
+  def index
+    @users = User.paginate(page: params[:page])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

**Pull request merged**